### PR TITLE
Adding _dd.iast.enabled flag to span during request end event

### PR DESCRIPTION
**Pull request by DDJavierSantos**
# What Does This Do
This PR adds a tag with the flag _dd.iast.enabled=1 to the span during the EndRequest event

# Motivation
This will allow the backend to filter traces where iast vulnerability detection was enabled.

# Additional Notes

### CODE REVIEW

File: dd-java-agent/iast/src/main/java/com/datadog/iast/NoopFlow.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package com.datadog.iast;
+
+import datadog.trace.api.gateway.Flow;
+
+public final class NoopFlow implements Flow<Void> {
```
Comment (smola) [inline dd-java-agent/iast/src/main/java/com/datadog/iast/NoopFlow.java:5]: I think you don't need this? This seems to be equivalent to `Flow.ResultFlow.empty()`

File: dd-java-agent/iast/src/test/groovy/com/datadog/iast/IastSystemTest.groovy [hunk 2]
```diff
@@ -1,10 +1,7 @@
 package com.datadog.iast
 
-
-import datadog.trace.api.gateway.Events
-import datadog.trace.api.gateway.InstrumentationGateway
-import datadog.trace.api.gateway.RequestContextSlot
-import datadog.trace.api.gateway.SubscriptionService
+import datadog.trace.api.TraceSegment
+import datadog.trace.api.gateway.*
```
Comment (smola) [inline dd-java-agent/iast/src/test/groovy/com/datadog/iast/IastSystemTest.groovy:4]: See https://github.com/DataDog/dd-trace-java/blob/master/CONTRIBUTING.md#intellij-idea

> Editor > Code Style > Java/Groovy > Imports
> Class count to use import with '*': 9999 (some number sufficiently large that is unlikely to matter)

File: dd-java-agent/iast/src/test/groovy/com/datadog/iast/IastSystemTest.groovy [hunk 3]
```diff
@@ -14,26 +11,33 @@ class IastSystemTest extends DDSpecification {
     final ig = new InstrumentationGateway()
     final ss = Spy(ig.getSubscriptionService(RequestContextSlot.IAST))
     final cbp = ig.getCallbackProvider(RequestContextSlot.IAST)
+    final traceSegment =Mock(TraceSegment)
```
Comment (smola) [inline dd-java-agent/iast/src/test/groovy/com/datadog/iast/IastSystemTest.groovy:14]:
```suggestion
    final traceSegment = Mock(TraceSegment)
```

File: dd-java-agent/iast/src/test/groovy/com/datadog/iast/IastSystemTest.groovy [hunk 4]
```diff
@@ -14,26 +11,33 @@ class IastSystemTest extends DDSpecification {
...
+    final igSpanInfo = Mock(IGSpanInfo)
+
 
     when:
     IastSystem.start(ss)
 
     then:
     1 * ss.registerCallback(Events.get().requestStarted(), _)
-    0 * _
+    1 * ss.registerCallback(Events.get().requestEnded(), _)
```
Comment (smola) [inline dd-java-agent/iast/src/test/groovy/com/datadog/iast/IastSystemTest.groovy:25]: Please, keep the `0 * _` as the last check in `then`. It helps you to spot unexpected interactions.

**Review by smola — CHANGES_REQUESTED** [review]

**Review by smola — APPROVED** [review]

**Review by manuel-alvarez-alvarez — APPROVED** [review]

**Pull request merged**